src/composeappmanager.cc

```diff
@@ -237,13 +237,22 @@ ComposeAppManager::AppsContainer ComposeAppManager::getAppsToUpdate(const Uptane
   return apps_to_update;
 }
 
-bool ComposeAppManager::checkForAppsToUpdate(const Uptane::Target& target) {
+std::vector<std::string> ComposeAppManager::checkForAppsToUpdate(const Uptane::Target& target) {
   cur_apps_to_fetch_and_update_ = getAppsToUpdate(target);
   if (!!cfg_.reset_apps) {
     cur_apps_to_fetch_ = getAppsToFetch(target);
   }
   are_apps_checked_ = true;
-  return cur_apps_to_fetch_and_update_.empty() && cur_apps_to_fetch_.empty();
+  std::vector<std::string> ret;
```

Comment:
Why do we need these two loops?

Reply:
To merge the app names from both cur_apps_to_fetch_and_update_ and cur_apps_to_fetch_. Next commit changes it anyway, perhaps I should squash the 2 last commits, if we decide that we want reason to be reported as well.

Reply:
Let's squash the commits.

Reply:
Commits squashed.